packages/sources/view-starknet-latest-answer/src/transport/function.ts

```diff
@@ -0,0 +1,106 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { AdapterResponse, makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/function'
+import { AdapterInputError } from '@chainlink/external-adapter-framework/validation/error'
+import { RpcProvider } from 'starknet'
+
+const logger = makeLogger('View Starknet Latest Answer')
+
+export type StarknetLatestAnswerFunctionTransportTypes = BaseEndpointTypes
+
+type RequestParams = typeof inputParameters.validated
+
+export class StarknetLatestAnswerFunctionTransport extends SubscriptionTransport<StarknetLatestAnswerFunctionTransportTypes> {
+  provider: RpcProvider | undefined
+
+  async initialize(
+    dependencies: TransportDependencies<StarknetLatestAnswerFunctionTransportTypes>,
+    adapterSettings: StarknetLatestAnswerFunctionTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<StarknetLatestAnswerFunctionTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>
+    try {
+      response = await this._handleRequestStarknet(param)
+    } catch (e: unknown) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterInputError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequestStarknet(
+    param: RequestParams,
+  ): Promise<AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>> {
+    const { address } = param
+
+    const networkEnvName = 'STARKNET_RPC_URL'
+
+    const rpcUrl = process.env[networkEnvName]
```

Comment:
This pattern of reading the environment variables directly was used in the `view-function-multi-chain` EA because the env var names wouldn't be known until runtime (because they use the `networkName` input param.

However here we know that this env var will always be named `STARKNET_RPC_URL`, so we can use the standard EA config for this. This will handle type-checking and default values & descriptions for us (allowing the README to be auto-generated). 

Here's an example of it in another EA: https://github.com/smartcontractkit/external-adapters-js/blob/main/packages/sources/finnhub/src/config/index.ts#L3-L25

Reply:
Once this is done you should be able to run `yarn generate:readme` and it will auto-update your README file